nemo/collections/nlp/models/dialogue/dialogue_zero_shot_slot_filling_model.py

```diff
@@ -0,0 +1,783 @@
+            one_predicted_slot_class = torch.zeros(len(one_mention_labels))
+
+            for idx, one_start_and_end in enumerate(start_and_end):
+                start, inclusive_end = one_start_and_end
+                for idy in range(start, inclusive_end + 1):
+                    one_predicted_slot_class[idy] = one_mention_labels[idx]
+            predicted_slot_class.append(one_predicted_slot_class)
+
+        predicted_slot_class = torch.stack(predicted_slot_class).to(bio_labels.device)
+        return predicted_slot_class
+
+    @staticmethod
+    def get_start_and_end_for_bio(one_bio_labels):
+        """
+        Getting the start and end of the mention from BIO(0, 1, 2) sequence.
+        Mention is start with B(with label 1) and follow by I(with label 2).
+        Args:
+            one_bio_labels: tensor with size (max_token_length, )
+            eg. [1, 0, 0, 1, 2]
+        
+        Returns:
+            start_and_end: list of mention's start and end. 
+            List length is the number of mentions in one_bio_labels.
+            eg. [[0, 0], [3, 4]]
+        """
+        start_and_end = []
+        last_seen = None
+        for i in range(len(one_bio_labels)):
+            if one_bio_labels[i] == 1:
+                if last_seen is not None:
+                    start_and_end[-1].append(last_seen)
+                start_and_end.append([i])
+                last_seen = i
+            elif one_bio_labels[i] == 2 and last_seen is not None:
+                last_seen = i
+            elif one_bio_labels[i] == 0:
+                if last_seen is not None:
+                    start_and_end[-1].append(last_seen)
+                last_seen = None
+        if last_seen is not None:
+            start_and_end[-1].append(last_seen)
+        return start_and_end
+
+    def get_entity_description_score(self, mention_hidden_states_pad):
+        """
+        Score entity description based on the dot product similarity between mention and description
+        Args: 
+            mention_hidden_states_pad: mention embedding of size (batch_size * max_num_mention * hidden_state_dim)
+            self.description_embeddings: description embeddings of size (num_slot_class * shared_embedding_dim)
+            
+        Returns:
+            dot product score matrix of given mention embedding and description embedding (batch_size * max_num_mention * num_slot_class)
+        
+        """
+        # Make sure that self.description_projection_mlp and self.description_embeddings are on the main trainer device
+        if next(self.description_projection_mlp.parameters()).device != self.device:
+            self.description_projection_mlp.to(self.device)
+        if self.description_embeddings.device != self.device:
+            self.description_embeddings = self.description_embeddings.to(self.device)
+
+        share_description_embedding = self.description_projection_mlp(self.description_embeddings)
```

Comment:
This naming is unclear, use something like projected_description_embedding and projected_mention_embedding